WordPress/gutenberg
# `CustomSelectControl`: add additional unit tests

## What?


Part of #55023

While looking into improving `CustomSelectControl`, the first stop is to refine the current tests and add additional ones. 

## Why?


Since `CustomSelectControl` isn't a native `select` element, it seems important to add tests that ensure it behaves the same as one. 

## How?


The additional tests are based on both `listbox` and `combobox` roles, the two appropriate roles for a `select`. 

## Notes 

After making changes to the component, the following tests/improvements could be implemented: 
- Replace `toHaveTextContent` with `toHaveValue`
- Ensure that HTML attributes valid for a `select` can be added (i.e. `disabled`)
- The default value or initially selected values will have focus when opening the popover, otherwise, it'll be the first element 
- Replace tests for experimental hint prop with tests for rendering any element

## Testing Instructions
Ensure tests pass: `npm run test:unit packages/components/src/custom-select-control/test/index.js`
